app/validators/capital_assessment/property_ownership_validator.rb

```diff
@@ -5,37 +5,46 @@ class PropertyOwnershipValidator < ActiveModel::Validator
     def validate(form)
       @form = form
 
-      form.errors.add :percentage_applicant_owned, error_key(form) unless valid_ownership_total?(form)
-
-      return unless form.include_partner_in_means_assessment? && !valid_ownership_total?(form)
-
-      form.errors.add :percentage_partner_owned, error_key(form)
+      other_owners? ? validate_other_owners : validate_ownership
     end
 
     private
 
-    def valid_ownership_total?(form)
-      percentage_ownerships = all_percentage_ownerships(form)
+    def validate_ownership
+      return if valid_ownership_total?
 
-      return percentage_ownerships.sum < 100 if other_owners?(form)
+      form.errors.add :percentage_applicant_owned, error_key
+      form.errors.add :percentage_partner_owned, error_key if partner_included?
+    end
 
-      percentage_ownerships.sum == 100
+    def validate_other_owners
+      form.errors.add :has_other_owners, error_key unless valid_ownership_total?
     end
 
-    def all_percentage_ownerships(form)
+    def valid_ownership_total?
+      total_percentages = all_percentage_ownerships.sum
+      other_owners? ? total_percentages < 100 : total_percentages == 100
+    end
+
+    def all_percentage_ownerships
       percentage_ownerships = []
       percentage_ownerships << form.percentage_applicant_owned unless form.percentage_applicant_owned.nil?
       percentage_ownerships << form.percentage_partner_owned unless form.percentage_partner_owned.nil?
 
       percentage_ownerships
     end
 
-    def other_owners?(form)
+    def partner_included?
+      form.include_partner_in_means_assessment?
+    end
+
+    def other_owners?
       form.has_other_owners&.yes?
     end
 
-    def error_key(form)
-      return :invalid_when_other_owners if other_owners?(form)
+    def error_key
+      return :invalid_when_other_owners if other_owners?
+      return :invalid_with_no_partner unless partner_included?
 
```

Comment:
This method might only be required for the `percentage_applicant_owned`  attribute when you are distinguishing between the error message when there is a partner vs. no partner. You could hardcode the error key for the other attributes